packages/astro/src/integrations/index.ts

```diff
@@ -38,6 +38,19 @@ async function withTakingALongTimeMsg<T>({
 	return result;
 }
 
+// Internally used to store instances of loggers.
+const Loggers = new Map<string, AstroIntegrationLogger>();
```

Comment:
I think this is being used to avoid creating a new AstroIntegrationLogger each time, is that right?

If so, this is leaking memory. Can you use `WeakMap<AstroIntegration, AstroIntegrationLogger>` instead?

Reply:
I wanted to use WeakMap in the first place, although it needs an object as key. I can use the whole integration object then